src/canmatrix/canmatrix.py

```diff
@@ -551,6 +552,48 @@ def pack_bitstring(length, is_float, value, signed):
 
     return bitstring
 
+@attr.s(cmp=False)
+class Arbitration_Id(object):
+    standard_id_mask = ((1 << 11) - 1)
+    extended_id_mask = ((1 << 29) - 1)
+    compound_extended_mask = (1 << 31)
+
+    id = attr.ib(default=None)
+    extended = attr.ib(default=None)
+
+    def __attrs_post_init__(self):
+        if self.extended is None or self.extended:
```

Comment:
```suggestion
        if self.extended:
```

Reply:
there was the requirement to get both arbitration ids - the extended and the basic one - with the "by_id" method. Therefore the "None" was. If you set extended to "None" and compare two arbitration ids you'll get a true for both - the extended and the basic version.

Reply:
Hmm...  maybe `by_id()` should be `return itertools.chain.from_iterable(by_id(id, extended) for type in (False, True))` or less niftily `return by_id(id, False) + by_id(id, True)`. (aside from the recursion in both cases...).  I'm really not sure.  I'm just taking to simpler classes at this point.  :|  Do what you want and we'll see how we feel about it later.